File: EchoesJava2/EchoesJava/src/renderingEngine/environment/EchoesSceneElement.java
package renderingEngine.src.environment;

import javax.media.opengl.GL2;
import renderingEngine.src.screenCanvas;

public class EchoesSceneElement
{
  protected boolean autoAdd = true;
  protected boolean fadingIn = false;
  protected boolean fadingOut = false;
  protected float transparancy = 1;
  protected int fadingFrames = 100;
  protected int id;
  
  public EchoesSceneElement(boolean autoAdd, boolean fadeIn, int framesToFade)
  {
    id = autoAdd ? screenCanvas.theCanvas.addSceneElement(this) : -1;
    fadingOut = false;
    fadingIn = fadeIn;
    fadingFrames = framesToFade;
    if(fadeIn)
      transparancy = 0;
    else
      transparancy = 1;
    }
          
    public void render(GL2 gl)
    {
      if (fadingIn)
      {
        transparancy += 1.0 / fadingFrames;
        if(transparancy >= 1.0)
        {
          transparancy = 1;
            fadingIn = false;
        }
      }
      else if (fadingOut)
      {
        transparancy -= 1.0 / fadingFrames;
        if(transparancy <= 0)
          remove(false, 100);
      }
      
      renderObj(gl);
    }
    
    public void renderObj(GL2 gl)
    {
    }
  
    public void remove(boolean fadeOut, int frames)
    {
      if(fadeOut)
      {
        fadingOut = true;
        fadingFrames = frames;
        transparancy = 1;
      }
      else
        screenCanvas.theCanvas.removeSceneElement(id);
    }
  }

File: EchoesJava2/EchoesJava/echoes-runtime/bin/rendering-engine/environment/EchoesSceneElement.py
'''
Created on 14 Oct 2009

@author: cfabric
'''

class EchoesSceneElement(object):
    '''
    classdocs
    '''


    def __init__(self, app, autoAdd=True, fadeIn = False, fadingFrames = 100):
        '''
        Constructor
        '''
        self.app = app
        if autoAdd:
            self.id = app.canvas.addSceneElement(self)
        else:
            self.id = -1

        self.fadingOut = False
        self.fadingIn = fadeIn
        self.fadingFrames = fadingFrames
        if fadeIn:
            self.transperancy = 0.0
        else:
            self.transperancy = 1.0

        
    def render(self):
        if hasattr(self, "fadingIn") and self.fadingIn:
            self.transperancy += 1.0/self.fadingFrames
            if self.transperancy >= 1.0:
                self.transperancy = 1.0
                self.fadingIn = False

        elif hasattr(self, "fadingOut") and self.fadingOut:
            self.transperancy -= 1.0/self.fadingFrames
            if self.transperancy <= 0.0:
                self.remove(False)
        
        self.renderObj()

    def renderObj(self):
        '''
        Overwrite this to render the element from the plugin system in EchoesGLCanvas
        '''
        print "Overwrite this method to render scene element"


    def remove(self, fadeOut=False, fadingFrames=100):
        if fadeOut:
            self.fadingOut = True
            self.fadingFrames = fadingFrames
            self.transperancy = 1.0
        else:
            self.app.canvas.removeSceneElement(self.id)
